src/main/java/org/schemaspy/service/TableService.java

```diff
@@ -441,6 +416,23 @@ protected long fetchNumRows(Database db, Table table, String clause, boolean for
         }
     }
 
+    private String getSchemaOrCatalog(Database db, Table table, boolean forceQuotes) throws SQLException {
+        String schemaOrCatalog = null;
+        if (table.getSchema() != null) {
+            schemaOrCatalog = table.getSchema();
+        } else if (table.getCatalog() != null) {
+            schemaOrCatalog = table.getCatalog();
+        }
+        if (schemaOrCatalog == null) {
+            return "";
+        }
+        if (forceQuotes) {
+            return db.quoteIdentifier(schemaOrCatalog) + ".";
```

Comment:
I reacted to it when I did the change. 
Mainly because getQuotedIdentifier might not return a quoted identifier.
That's why I wanted a quoteIdentifier method. Since it actually quotes the identifier.

I didn't change getQuotedIdentifier since I didnt feel quoteIdentifierIfNeeded was a good name.

I really have no good argument to not add the forceQuote to the getQuotedIdentifier other than why would I need to force it when I actually request a quoted identifier.

Reply:
@npetzall thank you for the change. The code is more clear for me. Unfortunately it is the conflict after some merges in your pull request.